packages/polymath-issuer/src/pages/compliance/components/EditApprovalForm.js

```diff
@@ -0,0 +1,250 @@
+// @flow
+
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { Modal, Button } from '@polymathnetwork/ui';
+import { withFormik } from 'formik';
+import { Form, Dropdown, DropdownItem } from 'carbon-components-react';
+import { RESTRICTION_TYPE } from '../../../constants';
+import {
+  bull,
+  Box,
+  PageCentered,
+  ContentBox,
+  Heading,
+  FormItem,
+  TextInput,
+  RadioInput,
+  PercentageInput,
+  NumberInput,
+  Grid,
+  FormItemGroup,
+  DatePickerInput,
+  TimePickerSelect,
+} from '@polymathnetwork/ui';
+import validator from '@polymathnetwork/ui/validator';
+import { editManualApproval } from '../../../actions/compliance';
+import {
+  validateTodayOrAfter,
+  validateDays,
+  validateExpiryTime,
+  REQUIRED_MESSAGE,
+  MORE_THAN_MESSAGE,
+  ADDRESS_MESSAGE,
+  MAX_DIGITS_MESSAGE,
+} from '../../sto/components/ConfigureSTOForm/validators'; // this is mangled put into global object
+import moment from 'moment';
+import { toWei } from '../../../utils/contracts';
+
+type Props = {
+  isOpen: boolean,
+  handleClose: () => any,
+};
+
+const formSchema = validator.object().shape({
+  date: validator.object().shape({
+    expiryDate: validator
+      .date()
+      .isRequired(REQUIRED_MESSAGE)
+      .test('validateExpiryDate', validateTodayOrAfter),
+    expiryTime: validator
+      .number()
+      .isRequired(REQUIRED_MESSAGE)
+      .test('validateExpiryTime', validateExpiryTime),
+  }),
+  fromAddress: validator
+    .string()
+    .isRequired(REQUIRED_MESSAGE)
+    .isAddress(ADDRESS_MESSAGE),
+  toAddress: validator
+    .string()
+    .isRequired(REQUIRED_MESSAGE)
+    .isAddress(ADDRESS_MESSAGE),
+  token: validator
+    .number()
+    .moreThan(0, MORE_THAN_MESSAGE)
```

Comment:
I believe `allowance` can be set to zero while modifying an existing approval.

Reply:
That is fine in the case that someone doesn't want to change their token value. Since this is incremental and not a hard value

Reply:
Just realized I changed how this works. We may need to restrict that part

Reply:
I don't follow. If one doesn't want to change the value, they'll leave it as `0`. But the validation above prevents that.

Reply:
They wouldn't leave it as 0. They would leave it as the value of the original. for eg, if a user has entered 100 Tokens, the edit form will fill it with 100 Tokens. It will mark the difference and select the correct adjustment for the token value